tests/runfiles/common.run

```diff
@@ -879,10 +879,8 @@ tests = ['zvol_cli_001_pos', 'zvol_cli_002_pos', 'zvol_cli_003_neg']
 tags = ['functional', 'zvol', 'zvol_cli']
 
 [tests/functional/zvol/zvol_misc]
-tests = ['zvol_misc_001_neg', 'zvol_misc_002_pos', 'zvol_misc_003_neg',
-    'zvol_misc_004_pos', 'zvol_misc_005_neg', 'zvol_misc_006_pos',
-    'zvol_misc_hierarchy', 'zvol_misc_rename_inuse', 'zvol_misc_snapdev',
-    'zvol_misc_volmode', 'zvol_misc_zil']
+tests = ['zvol_misc_002_pos', 'zvol_misc_hierarchy', 'zvol_misc_rename_inuse',
+    'zvol_misc_snapdev', 'zvol_misc_volmode', 'zvol_misc_zil']
```

Comment:
As a matter of documentation, let's make it clear why these still exist in the repository but are not referenced by the `common.run` file by adding them to an `illumos.run` file.

Reply:
@ikozhukhov can you confirm the output of `uname` on DilOS is "illumos"?

Reply:
I checked a DilOS installer in a VM and the uname is "SunOS" so I will name the file `sunos.run`

Reply:
Actually `uname -o` reports "illumos" so I'll update things to check that instead.

Reply:
Nevermind, that would be a problem with "GNU/Linux". It's simplest to just name the file `sunos.run` in this case.

Reply:
it'll be `dilos.run` in future :)

Reply:
i'll add later:
https://bitbucket.org/dilos/dilos-illumos/src/87795b2e11801e17c89487232355c16264e2bdcc/usr/src/test/zfs-tests/include/libtest.shlib#lines-96
when Brian will give me approve :)

Reply:
Sounds good.  Then for the moment, let's group these `sunos.run` as done in this PR.